core/src/models/Property.ts

```diff
@@ -373,6 +380,46 @@ export class Property extends LoggedModel<Property> {
     };
   }
 
+  // --- Cache Methods --- //
+
+  static async findAllWithCache(): Promise<Property[]> {
+    const now = new Date().getTime();
+    if (
+      _cachedProperties.expires > now &&
+      _cachedProperties.properties.length > 0
+    ) {
+      return _cachedProperties.properties;
+    }
+
+    _cachedProperties.properties = await Property.findAll();
+    _cachedProperties.expires = now + CACHE_TTL;
+    return _cachedProperties.properties;
+  }
+
+  static async findOneWithCache(key: string, value: string) {
+    const properties = await Property.findAllWithCache();
+    let property = properties.find((p) => p[key] === value);
+
+    if (!property) {
+      // fallback if not found
+      property = await Property.findOne({ where: { [key]: value } });
+      if (property) await Property.invalidateCache();
+    }
+
+    return property;
+  }
+
+  static invalidateLocalCache() {
+    _cachedProperties.expires = 0;
+  }
+
+  @AfterSave
+  @AfterUpdate
+  @AfterDestroy
+  static async invalidateCache() {
```

Comment:
technically, if this is in transaction, you want to do this after commit. otherwise, you'll clear on those other clusters opening up a window to refetch the wrong info just before this transaction runs,

Reply:
Should it do `invalidateLocalCache()` right here just to be sure or will `await` wait for that?

My gut says that `AfterSave`, `AfterDestroy` you would `invalidateLocalCache()` and then do a CLS sort of thing so that  `await redis.doCluster();` happens at the end of the transaction (and invliadate again just for good measure)

Reply:
Random note: At least in Rails, `AfterSave` is either of `AfterCreate` and `AfterUpdate` so the `AfterUpdate` here might be rendundant.

Reply:
Nice! I'll add the CLS wrap